.chloggen/non_cloud_hosting.yaml

```diff
@@ -0,0 +1,28 @@
+# Use this changelog template to create an entry for release notes.
+#
+# If your change doesn't affect end users you should instead start
+# your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the area of concern in the attributes-registry, (e.g. http, cloud, db)
+component: cloud, host
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
```

Comment:
On-premise is sometimes referred to as "private cloud".

Instead of making a breaking change by renaming the keys, could -- instead -- the definition/description of the existing keys simply be made more expansive such as changing the description/definition of `cloud.provider` to make it clear that on-prem environments can also be represented with that key (using a company-specific naming convention for their on-prem environments)?